elmPackages.elm-format: 0.8.7 -> 0.8.8

sandydoo opened the pull request:
This release officially supports GHC 9.6.
To avoid dependency conflicts with the elm compiler (see ansi-wl-pprint), elm-format is now built with GHC 9.8.
The GHC 9.4 elm package overrides have been removed.

~Derivations generated with the update scripts in the upstream repo.~
Derivations generated with `elm/update.sh`.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sandydoo committed:
elmPackages.elm-format: 0.8.7 -> 0.8.8

This release officially supports GHC 9.6.

To avoid dependency conflicts with the elm compiler (see ansi-wl-pprint), elm-format is now built with GHC 9.8.

The GHC 9.4 elm package overrides have been removed.

turboMaCk commented on pkgs/development/compilers/elm/packages/ghc9_4/default.nix:
```diff
@@ -1,53 +0,0 @@
-      elm-format-test-lib = fixHaddock (self.callPackage ./elm-format/elm-format-test-lib.nix { });
-      elm-format-markdown = fixHaddock (self.callPackage ./elm-format/elm-format-markdown.nix { });
-    };
-}
```
:tada:

turboMaCk commented:
@sandydoo are you sure this will conflict?

Otherwise change like this doesn't necessary need to go to haskell-updates. It has no impact on anything other from haskellPackages and doesn't depend on changes in the branch. So normal process is to merge these to master.

If there is a real danger of conflits that might change things.

sandydoo commented:
@sandydoo are you sure this will conflict?

I removed the ghc9_4 packages, since elm-format was the last one in there.

turboMaCk commented on pkgs/development/compilers/elm/packages/ghc9_8/elm-format/elm-format-markdown.nix:
```diff
@@ -23,6 +23,7 @@ mkDerivation {
     mtl
     text
   ];
+  doHaddock = false;
```
same as in other case

turboMaCk commented on pkgs/development/compilers/elm/packages/ghc9_8/elm-format/elm-format-lib.nix:
```diff
@@ -67,6 +67,7 @@ mkDerivation {
     text
   ];
   testToolDepends = [ tasty-discover ];
+  doHaddock = false;
```
here as well

turboMaCk commented on pkgs/development/compilers/elm/packages/ghc9_8/elm-format/avh4-lib.nix:
```diff
@@ -50,11 +49,11 @@ mkDerivation {
     text
   ];
   testToolDepends = [ tasty-discover ];
+  doHaddock = false;
```
and here

sandydoo commented on pkgs/development/compilers/elm/packages/ghc9_8/elm-format/elm-format-test-lib.nix:
```diff
@@ -56,6 +56,7 @@ mkDerivation {
     text
   ];
   testToolDepends = [ tasty-discover ];
+  doHaddock = false;
```
@turboMaCk it was generated by the upstream cabal2nix command.

I now see that  [`update.sh`](https://github.com/NixOS/nixpkgs/blob/69272053ce1b6bf20cad0a9421627ddc240a49bb/pkgs/development/compilers/elm/update.sh#L6-L10) has similar commands, but they weren't updated in the move from ghc 9.2 -> 9.4.

I can fix that and also add the `--no-haddock` option to the command to match upstream?

turboMaCk replied:
yes please do that.

sandydoo replied:
If `update.sh` is used to update elm-format, then I would need to revert the `postPatch` changes. There's a [confusing message](https://github.com/NixOS/nixpkgs/blob/a65c6c9fae542c37cff0f41947a72a5d748ec50b/pkgs/development/compilers/elm/packages/ghc9_4/default.nix#L11-L12) on the elm-format derivation that it's generated by the upstream script.

turboMaCk replied:
hmm ok. It seems this ship already sailed. Can you then remove it from update.sh? We can just maitain this manually going forward. I just don't want to confuse people.

turboMaCk replied:
maybe we can even remove that whole script. I think it's likely completely out of date.

sandydoo replied:
https://github.com/NixOS/nixpkgs/pull/419302/commits/e6ef81d433cb172b24d6905833b725d065b5e935

- I updated `update.sh`, ran it, and committed the changes. Other than some description changes, LGTM.
- I re-vendored the `postPatch`. This will make it easier to apply updates without checking out the upstream repository.

turboMaCk commented:
builds and works on x86_linux - updated in description

turboMaCk reviewed (approved):
LGTM

sandydoo committed:
elmPackages: update and run the `update.sh` script

- Vendored the post-patch script for elm-format.
- Updated `update.sh` and committed the changes.
